Event page tests

jgoley opened the pull request:
Adds a test scenario which tests each filtration section. Thoughts on adding a scenario which combines several filter properties?
Also adds:
- A clear button to clear all the filters
- Made the incident list reactive so when a new incident is added, it will reactively appear in the list
- Only add the location query if default range and selected not equal

[PT Chore](https://www.pivotaltracker.com/story/show/149213109)

nathanathan commented on packages/_tests/fixtures.coffee:
```diff
@@ -108,3 +132,24 @@ if Meteor.isAppTest
         console.warn("TestingAdmin user '#{email}' exists")
+
+    addIncidents: (eventId, incidentCount) ->
+      Meteor.call 'addEventSource', testSource2, eventId, (error, articleId) ->
```
Since these methods run on the server they can be called synchronously without a callback. Using async calls might cause timing issues in the tests since the addIncidents method will finish before the async calls complete.

jgoley commented:
@nathanathan Do you mind taking a look at this—I'm running into a few issues.
I went ahead and switched the date selector to the bootstrap version and in doing so am running into trouble with reactively setting the value of the input dates after a new incident is added.

jgoley commented:
The last commit fixes an issue which caused an infinite loop when the event page loaded (The dates being compared in the datepicker change event never matched up, i think because one was flagged as locale which caused a discrepancy?)
There are a few issues: 
For some reason the dates are updated just by clicking on the start date input. (i suppose this is mostly a UX issue)
During the tests, the datepicker seems to break—for some reason `selectedDateRange`  is not updated when the date input value changes. (even though it works fine otherwise)

nathanathan commented on client/controllers/slider.coffee:
```diff
@@ -10,44 +10,47 @@ Template.slider.onRendered ->
+
+    sliderEl.noUiSlider.on 'change', _.debounce (values, handle) =>
+      @data.selectedRange.set values
+      # Show or hide the left/right slider icon
```
This is already handled by reactivity in the render function. You don't need any of the jquery code here.

nathanathan commented on client/controllers/events/eventFiltration.coffee:
```diff
@@ -228,8 +241,24 @@ Template.eventFiltration.events
-      dates.push(new Date(input.value))
-    instance.selectedDateRange.set(dates)
+  'dp.change': (event, instance) ->
+    startStr = $('.start-date').data('DateTimePicker').date()?.utc().format('YYYY-MM-DD')
```
The date time picker displays locale dates. Interpreting them as utc could cause them to have different values that what the user entered.

nathanathan commented on client/controllers/events/eventFiltration.coffee:
```diff
@@ -3,9 +3,9 @@ import regionToCountries from '/imports/regionToCountries.json'
 formatDateForInput = (date) ->
   unless date
-    return moment.utc().format('YYYY-MM-DD')
+    return moment()
```
Why was this function modified?

jgoley replied:
I was trying to figure out why there is a discrepancy when the dates are compared in the picker event.

jgoley replied:
Looks like the start date is consistently a day off.

nathanathan commented:
selectedDateRange might not be updating during the tests because the date input never looses focus and triggers a change event. If that's not the issue making the tests use the calendar widget instead of entering text directly into the date field might work.

jgoley commented:
@nathanathan I switched to the daterangepicker for a slightly better UX and because it allows us a little more control in managing reactivity.
I also added a comparison of the selected date range with the event's default date range in the filter query autorun so we can keep the filterQuery reactiveVar empty when filters are not being used.
I also reverted back to updating the slider's max and min when the range changes rather than creating a new instance of the slider. When it is updated, the slider handle's movement is animated which I think is a better user experience.

nathanathan commented:
👍 Looks good
